controllers/workspace/http.go

```diff
@@ -29,11 +29,13 @@ var (
 )
 
 func setupHttpClients() {
-	transport := http.DefaultTransport.(*http.Transport).Clone()
-	healthCheckTransport := http.DefaultTransport.(*http.Transport).Clone()
-	healthCheckTransport.TLSClientConfig = &tls.Config{
+	tlsConfig := &tls.Config{
 		InsecureSkipVerify: true,
```

Comment:
@vinokurig but this means that we are skipping verification of TLS for every single request, no?

Reply:
@ibuziuk this is my understanding as well. Previously, we were only skipping TLS verification for the health check

Reply:
It might make more sense to add a DevWorkspace Operator Config field to allow disabling the TLS verification which users on an air-gapped cluster would toggle on.

Reply:
Added the property, @AObuchow please take a look.

Reply:
Overall looks good to me if we're going with the approach of enabling/disabling TLS verification (one small note in my next comment below about modifying the DevWorkspace Operator Config though).  I do think @tolusha's suggestion of using certificates retrieved from a secret is more optimal than my suggestion though. 

Additionally, we have to commit to any changes made to the DevWorkspace Operator Config since these are considered API changes, and we unfortunately can't easily remove any fields we add. So taking my simpler approach first, and then implementing @tolusha's suggestion later on would mean we'd have to keep `DisableTLSVerification` in the DevWorkspace Operator Config.

Thankfully, though both my suggestion and @tolusha's suggestion would resolve https://github.com/devfile/devworkspace-operator/issues/1248, they both provide different functionality and would not necessarily be redundantly overlapping each other if we were to support both fields in the DevWorkspace Operator Config. So implementing both approaches would not be terrible. @ibuziuk @tolusha @dkwon17 if any of you have any input on whether we should take a simpler approach first or not, please feel free to share.

Reply:
Regarding adding a new DevWorkspace Operator Config field, two areas need to be addressed:
1. Modifications need to be made to `mergeConfig()` and `GetCurrentConfigString()` in [sync.go](https://github.com/devfile/devworkspace-operator/blob/main/pkg/config/sync.go).

In [`mergeConfig()`](https://github.com/devfile/devworkspace-operator/blob/main/pkg/config/sync.go#L256-L272) we need to copy over the values of the new `DisableTLSVerification` field, something like: 


```diff
	if from.Routing != nil {
		if to.Routing == nil {
			to.Routing = &controller.RoutingConfig{}
		}
		if from.Routing.DefaultRoutingClass != "" {
			to.Routing.DefaultRoutingClass = from.Routing.DefaultRoutingClass
		}
		if from.Routing.ClusterHostSuffix != "" {
			to.Routing.ClusterHostSuffix = from.Routing.ClusterHostSuffix
		}
		if from.Routing.ProxyConfig != nil {
			if to.Routing.ProxyConfig == nil {
				to.Routing.ProxyConfig = &controller.Proxy{}
			}
			to.Routing.ProxyConfig = proxy.MergeProxyConfigs(from.Routing.ProxyConfig, defaultConfig.Routing.ProxyConfig)
		}
+		if from.Routing.DisableTLSVerification != nil {
+			to.Routing.DisableTLSVerification = from.Routing.DisableTLSVerification
+		}
	}
```


And in [`GetCurrentConfigString()`](https://github.com/devfile/devworkspace-operator/blob/main/pkg/config/sync.go#L463-L470) we need to declare if DisableTLSVerification is enabled. Something like:

```diff
	if routing != nil {
		if routing.ClusterHostSuffix != "" && routing.ClusterHostSuffix != defaultConfig.Routing.ClusterHostSuffix {
			config = append(config, fmt.Sprintf("routing.clusterHostSuffix=%s", routing.ClusterHostSuffix))
		}
		if routing.DefaultRoutingClass != defaultConfig.Routing.DefaultRoutingClass {
			config = append(config, fmt.Sprintf("routing.defaultRoutingClass=%s", routing.DefaultRoutingClass))
		}
+		if routing.DisableTLSVerification != nil && *routing.DisableTLSVerification {
+			config = append(config, "routing.DisableTLSVerification=true")
+		}
	}
```

You can run `make test` to make sure everything's working as expected. In particular, [this test](https://github.com/devfile/devworkspace-operator/blob/04f536990ebd114b4173bc3ba396e969f7863ebd/pkg/config/sync_test.go#L428-L441) should be passing.

2. The CRDs need to be re-generated since a new field was added to the DevWorkspace Operator Config. Run the following, and add the resulting changes into a separate commit: `make generate manifests fmt generate_default_deployment generate_olm_bundle_yaml`